backend/core/interactem/core/models/user.py

```diff
@@ -0,0 +1,66 @@
+from enum import Enum
+
+from pydantic import BaseModel
+
+from interactem.core.models.base import IdType
+
+UserOperatorID = IdType
+
+
+class UserOperatorInput(BaseModel):
+    label: str  # Human readable name of the input
+    description: str  # Human readable description of the input
+
+
+class UserOperatorOutput(BaseModel):
+    label: str  # Human readable name of the output
+    description: str  # Human readable description of the output
+
+
+class UserParameterType(str, Enum):
+    STRING = "str"
+    INTEGER = "int"
+    FLOAT = "float"
+    BOOLEAN = "bool"
+    MOUNT = "mount"
+    STR_ENUM = "str-enum"
+
+
+ParameterName = str
+
+
+class UserOperatorParameter(BaseModel):
+    name: ParameterName  # Name of the parameter
+    label: str  # Human readable name of the parameter
+    description: str  # Human readable description of the parameter
+    type: UserParameterType  # Type of the parameter
+    default: str  # Default value of the parameter
+    required: bool  # If the parameter is required
+    value: str | None = None  # Value of the parameter
+    options: list[str] | None = None  # List of options for STR_ENUM
+
+
+class UserOperatorTag(BaseModel):
+    value: str  # The actual tag value (e.g., "gpu", "ncem-4dstem")
+    description: str | None = None
+
+
+class ParallelType(str, Enum):
+    NONE = "none"
+    EMBARASSING = "embarrassing"
+
+
+class ParallelConfig(BaseModel):
+    type: ParallelType = ParallelType.NONE
+
+
+class UserOperator(BaseModel):
```

Comment:
What is a `UserOperator`?

Reply:
I will also add docstring, but it is the operator from when the user authors the operator